app/src/main/java/co/condorlabs/customcomponents/customcollapsibleview/CollapsibleView.kt

```diff
@@ -0,0 +1,555 @@
+            ?: resources.getString(R.string.show)
+        setTitle(typedArray.getString(R.styleable.CollapsibleView_collapsibleTitle))
+        setSubtitle(typedArray.getString(R.styleable.CollapsibleView_subtitle))
+        setActionLabelColor(
+            typedArray.getColor(
+                R.styleable.CollapsibleView_actionLabelColor,
+                NO_ID
+            )
+        )
+        setImageTint(typedArray.getColor(R.styleable.CollapsibleView_imageTintColor, NO_ID))
+        visibleIndicatorArrow(
+            typedArray.getBoolean(
+                R.styleable.CollapsibleView_visibleIndicatorArrow,
+                false
+            )
+        )
+
+        typedArray.recycle()
+    }
+
+    private fun collapseContent(requirementsHeight: Int) {
+        if (isCollapsedContent) {
+            collapseContent(ZERO, requirementsHeight, false)
+        } else {
+            collapseContent(requirementsHeight, ZERO, true)
+        }
+
+        isCollapsedContent = !isCollapsedContent
+
+        callOnCollapse(isCollapsedContent)
+    }
+
+    private fun collapseContent(from: Int, to: Int, isCollapsed: Boolean) {
+        updateActionLabel(isCollapsed)
+
+        val animator =
+            ValueAnimator.ofInt(from, to).setDuration(DEFAULT_COLLAPSIBLE_ANIMATION_TIME)?.apply {
+                addUpdateListener {
+                    updateContentHeight(it.animatedValue as Int)
+                    rotateIndicatorArrow(
+                        rotationIndicatorArrowValue(
+                            it.animatedValue as Int,
+                            if (from > to) from else to
+                        )
+                    )
+                }
+            } ?: return
+
+        startValueAnimator(animator)
+    }
+
+    private fun startValueAnimator(valueAnimator: ValueAnimator) {
+        AnimatorSet().apply {
+            play(valueAnimator)
+            interpolator = AccelerateDecelerateInterpolator()
+            start()
+        }
+    }
+
+    private fun updateContentHeight(height: Int = ZERO) {
+        flContent.apply {
+            layoutParams?.height = height
+            requestLayout()
+        }
+    }
+
+    private fun rotateIndicatorArrow(rotation: Float = ZERO_FLOAT) {
+        ivActionIndicator.rotation = -rotation
+    }
+
+    private fun rotationIndicatorArrowValue(currentValue: Int, limit: Int): Float {
+        return (COLLAPSIBLE_INDICATOR_ROTATION * ((currentValue * ONE_HUNDRED_FLOAT) / limit)) / ONE_HUNDRED_FLOAT
+    }
+
+    private fun updateActionLabel(isCollapsed: Boolean) {
+        tvActionLabel.text = if (isCollapsed) {
+            showActionLabel
+        } else {
+            hideActionLabel
+        }
+    }
+
+    private fun callOnCollapse(isCollapsed: Boolean): Boolean {
+        return onCollapseListener?.let { it.onCollapse(isCollapsed); true } ?: false
+    }
+
+    private fun Int.getMinimumCollapsibleContentPadding(): Int {
```

Comment:
this is like 
```
private static int getMinimumCollapsibleContentPadding(int this){
....
}
```
why do you want to create a private static method?, why do you want this method to be loaded into memory since the application is being launch